From fd40d9b6a80736cd76610ac6a562c4674f1f4565 Mon Sep 17 00:00:00 2001
From: Yaniv Levinsky <yaniv.levinsky@compulab.co.il>
Date: Thu, 14 Jun 2018 14:12:56 +0300
Subject: [PATCH 56/57] arm: imx7d: cl-som-imx7: fix USB reset

Fix the command 'usb reset'. The command fails with an error:
"USB not stopped: usbkbd still using USB"

This happens because usb_kbd_deregister() in common/usb_kbd.c supports
only kconfig definitions.

Move CONFIG_USB_KEYBOARD and CONFIG_SYS_STDIO_DEREGISTER flags from the
header file to kconfig.

Signed-off-by: Yaniv Levinsky <yaniv.levinsky@compulab.co.il>
---
 configs/cl-som-imx7_defconfig | 1 +
 include/configs/cl-som-imx7.h | 2 --
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/configs/cl-som-imx7_defconfig b/configs/cl-som-imx7_defconfig
index 7cb03488e0..a7b1b1d417 100644
--- a/configs/cl-som-imx7_defconfig
+++ b/configs/cl-som-imx7_defconfig
@@ -62,6 +62,7 @@ CONFIG_USB=y
 CONFIG_USB_EHCI_HCD=y
 CONFIG_MXC_USB_OTG_HACTIVE=y
 CONFIG_USB_STORAGE=y
+CONFIG_USB_KEYBOARD=y
 CONFIG_USB_GADGET=y
 CONFIG_CI_UDC=y
 CONFIG_OF_LIBFDT=y
diff --git a/include/configs/cl-som-imx7.h b/include/configs/cl-som-imx7.h
index 5763a668f5..93c5930311 100644
--- a/include/configs/cl-som-imx7.h
+++ b/include/configs/cl-som-imx7.h
@@ -212,9 +212,7 @@
 #define CONFIG_MXC_USB_PORTSC  (PORT_PTS_UTMI | PORT_PTS_PTW)
 #define CONFIG_MXC_USB_FLAGS   0
 #define CONFIG_USB_MAX_CONTROLLER_COUNT 2
-#define CONFIG_USB_KEYBOARD
 #define CONFIG_SYS_USB_EVENT_POLL_VIA_CONTROL_EP
-#define CONFIG_SYS_STDIO_DEREGISTER
 /* USB Networking options */
 #define CONFIG_USB_HOST_ETHER
 #define CONFIG_USB_ETHER_SMSC95XX
-- 
2.17.1

